consumer/consumer.go

```diff
@@ -222,57 +229,53 @@ func (consumer *StateSyncerConsumer) processNewEntriesInFile(isMempool bool) err
 				glog.Errorf("consumer.processNewEntriesInFile: Error reading next mempool entry from file: %s", err.Error())
 				break
 			}
-			return errors.Wrapf(err, "consumer.processNewEntriesInFile: Error reading next entry from file")
+			return revertTriggered, entriesProcessed, errors.Wrapf(err, "consumer.processNewEntriesInFile: Error reading next entry from file")
 		}
 		if fileEOF {
 			break
 		}
+		entriesProcessed = true
 		if !isMempool {
-			if err = consumer.SyncCommittedEntry(stateChangeEntry); err != nil {
-				return errors.Wrapf(err, "consumer.processNewEntriesInFile: Error syncing committed entry")
+			entryRevertTriggered, err := consumer.SyncCommittedEntry(stateChangeEntry)
+			// Update the overall revertTriggered flag if this entry triggered a revert
+			revertTriggered = revertTriggered || entryRevertTriggered
+			if err != nil {
+				return revertTriggered, entriesProcessed, errors.Wrapf(err, "consumer.processNewEntriesInFile: Error syncing committed entry")
 			}
 		} else {
-			if err = consumer.SyncMempoolEntry(stateChangeEntry); err != nil {
-				return errors.Wrapf(err, "consumer.processNewEntriesInFile: Error syncing mempool entry")
+			entryRevertTriggered, err := consumer.SyncMempoolEntry(stateChangeEntry)
+			// Update the overall revertTriggered flag if this entry triggered a revert
+			revertTriggered = revertTriggered || entryRevertTriggered
+			if err != nil {
+				return revertTriggered, entriesProcessed, errors.Wrapf(err, "consumer.processNewEntriesInFile: Error syncing mempool entry")
```

Comment:
seems like we could dedupe the code a little bit here, but I'm ok w/ whatever you want to do. I would define the `entryRevertTriggered` bool before the if/else and then do the following (having just the `SyncCommittedEntry` or `SyncMempoolEntry` and setting a string called `entryType` to `committed` or `mempool` for use in the error msg in the if/else)
```
		revertTriggered = revertTriggered || entryRevertTriggered
		if err != nil {
				return revertTriggered, entriesProcessed, errors.Wrapf(err, "consumer.processNewEntriesInFile: Error syncing %v entry", entryTryp)
```

Reply:
Good thought - done.